src/services/expense.service.js

```diff
@@ -0,0 +1,74 @@
+const { Op } = require('sequelize');
+const { Expense } = require('../models/Expense.model.js');
+
+const getAllExpense = async ({ userId, categories, from, to }) => {
+  const params = {};
+
+  if (userId) {
+    params.userId = userId;
+  }
+
+  if (categories) {
+    if (Array.isArray(categories)) {
+      params.category = { [Op.in]: categories };
+    } else {
+      params.category = categories;
+    }
+  }
+
+  if (from || to) {
+    params.spentAt = {};
+
+    if (from) {
+      params.spentAt[Op.gte] = from;
+    }
+
+    if (to) {
+      params.spentAt[Op.lte] = to;
+    }
+  }
+
+  const expList = Expense.findAll({ where: params });
+
+  return expList;
+};
+
+const getExpenseById = async (id) => {
+  return Expense.findByPk(id);
+};
+
+const createExpense = async (data) => {
+  return Expense.create(data);
```

Comment:
Consider adding validation or error handling to ensure that the data being created is valid and to handle any potential errors that might occur during the creation process.